pkg/controller/direct/compute/targettcpproxy/targettcpproxy_controller.go

```diff
@@ -258,13 +260,18 @@ func (a *targetTCPProxyAdapter) Update(ctx context.Context, updateOp *directbase
 	if err != nil {
 		return fmt.Errorf("get ComputeTargetTCPProxy parent %s: %w", a.id.External, err)
 	}
-	region := parent.Region
+	location := parent.Location
+
+	// Regional API does not support Update
+	if location != "global" {
+		return k8s.NewUpdateNotSupportedError(a.desired.GroupVersionKind(),
```

Comment:
Can I ask why we want to define a new structed type?

Reply:
I feel that could be a common thing we want to handle as we migrate more resources to direct. Did a quick [code search](https://github.com/search?q=repo%3AGoogleCloudPlatform%2Fk8s-config-connector%20skipUpdate&type=code) and found several "skipUpdate" due to no API support. We used "immutable fields" to prevent update, but since we removed immutability webhook for direct resources, the CEL rule has limit guarantee on only required fields. I feel might be better to create a new error type, in case we have more similar cases in the future. But I'm open to suggestions.

Reply:
Also worth mentioning that regional TCP Proxy is a new direct resource, so luckily we won't need to worry about backward compatibility. I look forward to getting users feedback from this resource, and explore how to handle "skipUpdate" for other resources.  (For existing resources, we'll need to consider backward compatibility carefully.)

Reply:
We could, but maybe not in this PR. Because this new error in k8s does not affect the reconcile lifecycle. If you want to use a generic error, we should not put it inside k8s. Besides, the error message needs to be more carefully designed since this is user facing and reflects the reconciliation difference.